packages/osd-plugin-helpers/src/tasks/update_versions.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// @ts-ignore
+import { readFile, writeFile } from 'fs/promises';
+import path from 'path';
+import { createFailError } from '@osd/dev-utils';
+import { FileUpdateContext, ObjectUpdateContext, VersionContext } from '../contexts';
+
+export async function updateVersions({
+  log,
+  sourceDir,
+  newVersion,
+}: VersionContext): Promise<boolean | undefined> {
+  if (!newVersion) throw createFailError('Missing new version value');
+  if (!/^\d+\.\d+\.\d+/.test(newVersion))
+    throw createFailError('The new version is not a valid semantic version');
+
+  const cleanVersion = newVersion.replace(/^(\d+\.\d+\.\d+)(-.*)?$/, '$1');
+
+  const updateManifestFile = updateFile({
+    log,
+    file: path.join(sourceDir, 'opensearch_dashboards.json'),
+    updates: {
+      version: `${cleanVersion}.0`,
```

Comment:
Why to we want to always want to force `.0` as the suffix, even if the user has specified a different, valid one?

Reply:
@kavilla, I updated this PR to handle that scenario.